js/filtration.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+(function () {
+
+  var filterBlock = document.querySelector('.img-filters');
+  var filterButtons = filterBlock.querySelectorAll('.img-filters__button');
+
+  var filterNotActivated = true;
+  var NUMBER_OF_NEW_IMAGES = 10;
+
+  function filtration() {
+    if (filterNotActivated) {
+      var defaultPicturesKit = window.usersPictures;
+      filterNotActivated = false;
+    }
+
+    filterBlock.classList.remove('img-filters--inactive');
+
+    filterButtons.forEach(function (filterButton, filterIndex) {
+      filterButton.addEventListener('click', onFilterSelect);
+
+      function onFilterSelect() {
+        switch (filterIndex) {
+          case 0:
+            window.usersPictures = defaultPicturesKit;
+            break;
+          case 1:
+            window.usersPictures = getNewPictures(defaultPicturesKit.slice());
+            break;
+          case 2:
+            window.usersPictures = getDiscussedPictures(defaultPicturesKit.slice());
+            break;
+        }
+        clearActiveFilterButton();
+        filterButton.classList.add('img-filters__button--active');
+        window.debounce(drawFilteredPictures);
+      }
+    });
+  }
+
+  function drawFilteredPictures() {
+    var pictures = document.querySelectorAll('.picture');
+
+    pictures.forEach(function (picture) {
+      document.querySelector('.pictures').removeChild(picture);
+    });
+
+    window.drawPictures(window.usersPictures);
+    window.preview();
+  }
+
+  function getNewPictures(allPictures) {
+    var indexNewPicture;
+    var newPictures = [];
+
+    for (var i = 0; i < NUMBER_OF_NEW_IMAGES; i++) {
+      indexNewPicture = window.util.getIndexFromRange(0, allPictures.length);
+      newPictures.push(allPictures[indexNewPicture]);
```

Comment:
```suggestion
      newPictures.push(allPictures.splice(indexNewPicture, 1)[0]);
```

а третью строчку убрать